h/services/feature.py

```diff
@@ -0,0 +1,124 @@
+# -*- coding: utf-8 -*-
+
+from __future__ import unicode_literals
+
+import re
+
+from h import models
+from h.util.db import lru_cache_in_transaction
+
+PARAM_PATTERN = re.compile(r'\A__feature__\[(?P<featurename>[A-Za-z0-9_-]+)\]\Z')
+
+
+class UnknownFeatureError(Exception):
+    pass
+
+
+class FeatureRequestProperty(object):
+    """
+    Helper object for accessing feature flags.
+
+    An instance of FeatureRequestProperty is available on the request object
+    as ``request.feature`` in order to simplify access to the feature service.
+    """
+
+    def __init__(self, request):
+        self.request = request
+        self.svc = request.find_service(name='feature')
+
+    def __call__(self, name):
+        """Get the status of feature flag `name` for the current user."""
+        return self.svc.enabled(name, user=self.request.authenticated_user)
+
+    def all(self):
+        """Get the status of all feature flags for the current user."""
+        return self.svc.all(user=self.request.authenticated_user)
+
+
+class FeatureService(object):
+    """
+    Manages access to feature flag status.
+
+    This service is exposed on the request object as `request.feature` for
```

Comment:
Is this part of the docstring supposed to be here, or does it predate you introducing the helper class?